lib/src/algorithms.dart

```diff
@@ -176,8 +176,8 @@ void mergeSort<T>(List<T> list,
   _mergeSort<T>(list, compare, middle, end, scratchSpace, 0);
   var firstTarget = end - firstLength;
   _mergeSort<T>(list, compare, start, middle, list, firstTarget);
-  _merge<T>(compare, list, firstTarget, end, scratchSpace, 0, secondLength, list,
-      start);
+  _merge<T>(compare, list, firstTarget, end, scratchSpace.cast<T>(), 0,
```

Comment:
I would initialize scratchSpace to `List<T>.filled(secondLength, list[start]);`.

Since we know that  0 &le; `start` &lt; `start` + `_MERGE_SORT_LIMIT` &le; `end` &le; `list.length`, that value exist.
(Or we would have known if the function had started with the appropriate:
```dart
   end = RangeError.checkValidRange(start, end, list.length);
```
so consider adding that while you are here).

I'd make the parameter in `_mergeSort` and `_merge` below a `List<T>` as well.

Reply:
done - that worked thanks!